### PR TITLE
Add orthogonalization

### PR DESCRIPTION
**Pull request by hugorichard**
This PR allows a user to set ortho=True.

### CODE REVIEW

File: qndiag/tests/test_qndiag.py [hunk 1]
```diff
@@ -39,3 +39,19 @@ def test_errors():
...
         x[0] = np.array([[0, 1], [1, 0]])
         qndiag(x)
+
+@pytest.mark.parametrize('weights', [None])
+def test_qndiag_ortho(weights):
+    n, p = 10, 3
+    rng = np.random.RandomState(42)
+    diagonals = rng.uniform(size=(n, p))
+    A = rng.randn(p, p)  # mixing matrix
+    Ua,_, Va = np.linalg.svd(A, full_matrices=False)
```
Comment (agramfort) [inline qndiag/tests/test_qndiag.py]:
```suggestion
    Ua, _, Va = np.linalg.svd(A, full_matrices=False)
```

File: qndiag/tests/test_qndiag.py [hunk 2]
```diff
@@ -39,3 +39,19 @@ def test_errors():
         x += x.swapaxes(1, 2)
         x[0] = np.array([[0, 1], [1, 0]])
         qndiag(x)
+
+@pytest.mark.parametrize('weights', [None])
```
Comment (agramfort) [inline qndiag/tests/test_qndiag.py]: I would have just use a parametrize with ortho too and inside if ortho you simulate with an orthogonal matrix and you make sure the returned matrix is orthogonal.

it avoids some code dupe between tests

File: qndiag/qndiag.py [hunk 3]
```diff
@@ -128,22 +134,29 @@ def qndiag(C, B0=None, weights=None, max_iter=1000, tol=1e-6,
...
+                    weights=weights_, axis=0)
+        if ortho:
+            det = h + h.T - 1
+            det[det < lambda_min] = lambda_min  # Regularize
+            direction = -G / det
+        else:
+            # Quasi-Newton's direction
+            det = h * h.T - 1.
+            det[det < lambda_min] = lambda_min  # Regularize
+            direction = -(G * h.T - G.T) / det
```
Comment (agramfort) [inline qndiag/qndiag.py:155]:
```suggestion
        det = h + h.T - 1
        det[det < lambda_min] = lambda_min  # Regularize
        if ortho:
            direction = -G / det
        else:
            direction = -(G * h.T - G.T) / det
```

**Review by agramfort — COMMENTED** [review]
we should add a changlog file or a what's new page on the doc and make a new release :)

File: qndiag/qndiag.py (hunk 3 above)
Comment (hugorichard) [reply qndiag/qndiag.py:155]: Wait it is not the same thing there. you h + h.T - 1 or h * h.T - 1 but I see the spirit

**Comment by agramfort** [comment]
Sorry read too fast 💨

File: qndiag/tests/test_qndiag.py [hunk 4]
```diff
@@ -1,25 +1,30 @@
...
 @pytest.mark.parametrize('weights', [None, True])
-def test_qndiag(weights):
+@pytest.mark.parametrize('ortho', [False, True])
+def test_qndiag(weights, ortho):
     n, p = 10, 3
     rng = np.random.RandomState(42)
     diagonals = rng.uniform(size=(n, p))
     A = rng.randn(p, p)  # mixing matrix
+    if ortho:
+        Ua,_, Va = np.linalg.svd(A, full_matrices=False)
```
Comment (agramfort) [inline qndiag/tests/test_qndiag.py]:
```suggestion
        Ua, _, Va = np.linalg.svd(A, full_matrices=False)
```

**Review by agramfort — COMMENTED** [review]
Can you add a what’s new page to the doc like we do for our projects?

**Comment by agramfort** [comment]
Can you also see why CIs fail?

File: qndiag/tests/test_qndiag.py [hunk 5]
```diff
@@ -1,25 +1,30 @@
 import numpy as np
-from numpy.testing import assert_array_equal
+from numpy.testing import assert_array_equal, assert_allclose
```
Comment (pierreablin) [inline qndiag/tests/test_qndiag.py]: assert_array_equal is now unused

**Comment by agramfort** [comment]
ok to merge @hugorichard @pierreablin !

**Comment by hugorichard** [comment]
Thanks Alex ! Only Pierre can merge I think.

**Comment by agramfort** [comment]
thx @hugorichard

**Pull request merged**